src/routes/PublicLayoutRouter/PublicLayoutRouter.js

```diff
@@ -1,77 +1,195 @@
+import { useParams, useLocation, useHistory, Redirect, Route, Switch } from 'react-router-dom'
 import Div100vh from 'react-div-100vh'
-import { matchPath, Redirect, Route, Switch } from 'react-router-dom'
+import { POST_DETAIL_MATCH, GROUP_DETAIL_MATCH } from 'util/navigation'
+import { CENTER_COLUMN_ID, DETAIL_COLUMN_ID } from 'util/scrolling'
 import HyloCookieConsent from 'components/HyloCookieConsent'
-import TopicSupportComingSoon from 'components/TopicSupportComingSoon'
 import GroupDetail from 'routes/GroupDetail'
 import GroupExplorer from 'routes/GroupExplorer'
+import Loading from 'components/Loading'
 import MapExplorer from 'routes/MapExplorer'
 import PostDetail from 'routes/PostDetail'
-import { OPTIONAL_POST_MATCH, POST_DETAIL_MATCH, GROUP_DETAIL_MATCH } from 'util/navigation'
-import { CENTER_COLUMN_ID, DETAIL_COLUMN_ID } from 'util/scrolling'
 import './PublicLayoutRouter.scss'
+import gql from 'graphql-tag'
 
 export default function PublicLayoutRouter (props) {
-  const { location } = props
+  const routeParams = useParams()
+  const location = useLocation()
+  const isMapView = routeParams?.view === 'map'
+
+  return (
+    <Div100vh styleName={cx('public-container', { 'map-view': isMapView })}>
+      <PublicPageHeader />
+      <Switch>
+        <Route path={`/${POST_DETAIL_MATCH}`} exact component={PublicPostDetail} />
+        <Route path='/:context(groups)/:groupSlug' exact component={PublicGroupDetail} />
+        <Route path='/:context(public)/:view(map)' component={MapExplorerLayoutRouter} />
+        <Route path='/:context(public)/:view(groups)' exact component={GroupExplorerLayoutRouter} />
+        {/* Remove this once we show the public stream */}
+        <Redirect exact from={`/public/${POST_DETAIL_MATCH}`} to='/post/:postId' />
+        <Redirect to={{ pathname: '/public/map', state: { from: location } }} />
+      </Switch>
+      <HyloCookieConsent />
+    </Div100vh>
+  )
+}
+
+export function PublicGroupDetail (props) {
+  const dispatch = useDispatch()
+  const routeParams = useParams()
+  const history = useHistory()
+  const [loading, setLoading] = useState(true)
+  const groupSlug = routeParams?.groupSlug
+  const checkIsPublicGroup = groupSlug => {
+    return {
+      type: 'IS_GROUP_PUBLIC',
+      graphql: {
+        query: gql`
+          query CheckIsGroupPublic ($slug: String) {
+            group (slug: $slug) {
+              visibility
+            }
+          }
+        `,
+        variables: { slug: groupSlug }
+      },
+      meta: { extractModel: 'Group' }
+    }
+  }
+
+  useEffect(() => {
+    (async () => {
+      setLoading(true)
 
-  const pathMatchParams = useMemo(() => (
-    matchPath(location.pathname, [
-      '/:context(public)/:view(groups|map)?'
-    ])?.params || { context: 'public' }
-  ), [location.pathname])
+      const result = await dispatch(checkIsPublicGroup(groupSlug))
+      const isPublicGroup = result?.payload?.data?.group?.visibility === 2
```

Comment:
won't this error out if you cant see the group, or just return null? seems like you dont need this check.

Reply:
Sorry, strike that. I see what you mean @tibetsprague. If the query returns anything then its implicitly Public I guess. So yeah, you shouldn't also need to check for the visibility status at this point.

Reply:
No, it doesn't need it at all, its just being emphatic